plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -1676,6 +1646,111 @@ public static function log( $message, $level = 'info' ) {
 
 		self::$log->log( $level, $message, array( 'source' => 'helper' ) );
 	}
+
+	/**
+	 * Handles WC Helper disconnect tasks.
+	 *
+	 * @return void
+	 */
+	public static function disconnect() {
+		WC_Helper_API::post(
+			'oauth/invalidate_token',
+			array(
+				'authenticated' => true,
+			)
+		);
+
+		WC_Helper_Options::update( 'auth', array() );
+		WC_Helper_Options::update( 'auth_user_data', array() );
+
+		self::_flush_subscriptions_cache();
+		self::_flush_updates_cache();
+	}
+
+	/**
+	 * Checks if `access_token` exists in `auth` option.
+	 *
+	 * @return bool
+	 */
+	public static function is_site_connected(): bool {
+		$auth = WC_Helper_Options::get( 'auth' );
+
+		// If `access_token` is empty, there's no active connection.
+		return ! empty( $auth['access_token'] );
+	}
+
+	/**
+	 * Allows to connect with WCCOM using application password. used it to connect via CLI
+	 *
+	 * @param string $password The application password
+	 *
+	 * @return void|WP_Error
+	 */
+	public static function connect_with_password( string $password ) {
+		$request = WC_Helper_API::post(
+			'connect',
+			array(
+				'headers' => array( 'X-API-Key' => $password ),
+				'body'    => wp_json_encode( array( 'home_url' => home_url() ) ),
+			)
+		);
+
+		$code = wp_remote_retrieve_response_code( $request );
+
+		if ( 403 === $code ) {
+			$message = 'Invalid password';
+			self::log( $message );
+
+			return new WP_Error( 'connect-with-password-invalid-password', $message );
+		} elseif ( 200 !== $code ) {
+			$message = sprintf( 'Call to /connect returned a non-200 response code (%d)', $code );
+			self::log( $message );
+
+			return new WP_Error( 'connect-with-password-' . $code, $message );
+		}
+
+		$access_data = json_decode( wp_remote_retrieve_body( $request ) );
```

Comment:
Needs to be updated to avoid fatal error:
```
$access_data = json_decode( wp_remote_retrieve_body( $request ), true );
```

Reply:
Fixed in 885b0de